src/js/src/vlm_pipeline/vlm_pipeline_wrapper.cpp

```diff
@@ -0,0 +1,287 @@
+        VALIDATE_ARGS_COUNT(info, 4, "init()");
+        const std::string model_path = js_to_cpp<std::string>(env, info[0]);
+        const std::string device = js_to_cpp<std::string>(env, info[1]);
+        const auto& properties = js_to_cpp<ov::AnyMap>(env, info[2]);
+        OPENVINO_ASSERT(info[3].IsFunction(), "init callback is not a function");
+        Napi::Function callback = info[3].As<Napi::Function>();
+
+        VLMInitWorker* asyncWorker =
+            new VLMInitWorker(callback, this->pipe, this->is_initializing, model_path, device, properties);
+        asyncWorker->Queue();
+    } catch (const std::exception& ex) {
+        Napi::Error::New(env, ex.what()).ThrowAsJavaScriptException();
+    }
+    return env.Undefined();
+}
+
+Napi::Value VLMPipelineWrapper::generate(const Napi::CallbackInfo& info) {
+    auto env = info.Env();
+    try {
+        OPENVINO_ASSERT(this->pipe, "VLMPipeline is not initialized");
+        OPENVINO_ASSERT(!*this->is_generating, "Another generation is already in progress");
+        *this->is_generating = true;
+        VALIDATE_ARGS_COUNT(info, 6, "generate()");
+        VLMTsfnContext* context = nullptr;
+
+        // Arguments: prompt, images, videos, streamer, generationConfig, callback
+        auto prompt = js_to_cpp<std::string>(env, info[0]);
+        auto images = js_to_cpp<std::vector<ov::Tensor>>(env, info[1]);
+        auto videos = js_to_cpp<std::vector<ov::Tensor>>(env, info[2]);
+        OPENVINO_ASSERT(info[3].IsFunction() || info[3].IsUndefined(), "generate callback is not a function");
+        auto streamer = info[3].As<Napi::Function>();
+        auto generation_config = js_to_cpp<ov::AnyMap>(env, info[4]);
+        OPENVINO_ASSERT(info[5].IsFunction(), "generate callback is not a function");
+        auto callback = info[5].As<Napi::Function>();
+
+        context = new VLMTsfnContext(prompt, this->is_generating);
+        context->images = std::move(images);
+        context->videos = std::move(videos);
+        context->pipe = this->pipe;
+        context->generation_config = std::make_shared<ov::AnyMap>(generation_config);
+
+        context->callback =
+            Napi::ThreadSafeFunction::New(env,
+                                          callback,                     // JavaScript function called asynchronously
+                                          "VLM_generate_callback",      // Name
+                                          0,                            // Unlimited queue
+                                          1,                            // Only one thread will use this initially
+                                          [context, this](Napi::Env) {  // Finalizer used to clean threads up
+                                              context->native_thread.join();
+                                              delete context;
+                                          });
```

Comment:
This Finalizer is repeated for each pipeline. We can extract it like here and reuse it: https://github.com/openvinotoolkit/openvino/blob/805d41a1c7ae209619651e57863d961ee32c65c2/src/bindings/js/node/src/infer_request.cpp#L197